### PR TITLE
fix: The monster animation does not go to idle state once the puzzle is completed (FM-345)

### PR DESCRIPTION
**Pull request by woodsjohn-codev**
# Changes
- Fix the monster animation doest not go idle state once the puzzle is completed
- Merge monster to rive-monster-component
- Deleted the monster file
- Add test cases

# How to test
- Open the start screen of FTM.
- Tap or click any part of the start screen to proceed to the next screen
- Select any level
- Play the game
- Make sure the monster go to idle state every puzzle loaded
- Make sure monster will work properly in levelend and start scene

Ref: [FM-345](https://curiouslearning.atlassian.net/browse/FM-345)

[FM-345]: https://curiouslearning.atlassian.net/browse/FM-345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/scenes/start-scene/start-scene.ts [hunk 1]
```diff
@@ -68,7 +67,8 @@ export class StartScene {
       height: this.canavsElement.height,
       onLoad: () => {
         this.riveMonster.play(RiveMonsterComponent.Animations.IDLE); // Start with the "Idle" animation
-      }
+      },
+      gameCanvas: this.canvas
```
Comment (bernhardccodev) [inline src/scenes/start-scene/start-scene.ts:71]: Start scene doesn't need the hitbox. Make this optional.

Comment (woodsjohn-codev) [reply src/scenes/start-scene/start-scene.ts]: @bernhardccodev thanks i applied an adjustment for this.

**Review by bernhardccodev — APPROVED** [review]

File: src/components/riveMonster/rive-monster-component.ts [hunk 2]
```diff
@@ -1,70 +1,154 @@
...
+    OPENING_MOUTH_EAT: 'Opening Mouth Eat',
+    EAT_HAPPY: 'Eat Happy',
+    IDLE: 'Idle',
+    EAT_DISGUST: 'Eat Disgust',
+  };
+
+  constructor(props: RiveMonsterComponentProps) {
+    this.props = props;
+
+    if (this.props.gameCanvas) {
```
Comment (bernhardccodev) [inline src/components/riveMonster/rive-monster-component.ts:42]: Add a comment regarding the this.props.gameCanvas will not be needed once FM-333 pixelate issue fix has been applied.

Comment (woodsjohn-codev) [reply src/components/riveMonster/rive-monster-component.ts:43]: Thanks added a todo comment

**Review by ashwinnair-chimple — APPROVED** [review]
Approved

**Pull request merged**